.pre-commit-config.yaml

```
@@ -0,0 +1,23 @@
+exclude: |
+  (?x)^(
+      build|
+      external_libs|
+      lightgbm-python|
+      lightgbm_r|
+  )$
+
+repos:
+  - repo: https://github.com/astral-sh/ruff-pre-commit
+    # Ruff version.
+    rev: v0.2.1
+    hooks:
+      # Run the linter.
+      - id: ruff
+        args: ["--config", "pyproject.toml"]
```

I think this is optional. Instead, I would maybe add configuration to autofix auto-fixable issues:
```suggestion
        args: ["--fix", "--exit-non-zero-on-fix"]
```

---

I agree that it's optional, and `pyproject.toml` is picked up automatically. Leaving this in was a mistake. I think I'd had that at some point while trying to figure out how to have different `exclude` patterns for `ruff check` vs. `ruff format`.

I'll remove this.

However... I don't support adding `--fix` for `ruff check`. I'd rather have the opportunity to decide for myself whether to change something or to ignore it like this:

https://github.com/microsoft/LightGBM/blob/cc733f8595267f313886f92ed5d1285010ba8f3f/python-package/pyproject.toml#L131-L145

I've worked in other projects with `--fix` and found it annoying when `pre-commit` undoes my not-yet-committed work in significant ways (like removing variables that appear "unused" but which I just haven't yet finished the implementation for).

I think we should omit `--fix`.

---

I just pushed https://github.com/microsoft/LightGBM/pull/6308/commits/4f1a3d3f154820f23f93a52e0d8503baecefc138 removing this unnecessary `--config`, thanks for catching that!

---

> I think we should omit `--fix`.

Fine for me ;)